File: frontend/demo/component/datetimepicker/date-time-picker-default-time.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/date-time-picker';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('date-time-picker-default-time')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-date-time-picker label="Appointment" default-time="09:00"></vaadin-date-time-picker>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/datetimepicker/DateTimePickerDefaultTime.java
package com.vaadin.demo.component.datetimepicker;

import com.vaadin.flow.component.datetimepicker.DateTimePicker;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-source-line

import java.time.LocalTime;

@Route("date-time-picker-default-time")
public class DateTimePickerDefaultTime extends Div {

    public DateTimePickerDefaultTime() {
        // tag::snippet[]
        DateTimePicker dateTimePicker = new DateTimePicker("Appointment");
        dateTimePicker.setDefaultTime(LocalTime.of(9, 0));
        // end::snippet[]
        add(dateTimePicker);
    }

    public static class Exporter extends // hidden-source-line
            DemoExporter<DateTimePickerDefaultTime> { // hidden-source-line
    } // hidden-source-line
}
